lib/network/validator_connection_manager.go

```diff
@@ -168,15 +174,120 @@ func (c *ValidatorConnectionManager) connectingValidator(v *node.Validator) {
 		if c.setConnected(v, err == nil) {
 			if err == nil {
 				c.log.Debug("validator is connected", "validator", v.Address())
+				if c.countConnectedUnlocked() < c.policy.Threshold() {
+					continue
+				}
+
+				if !c.connectedEqualOrOverThreshold {
+					c.updateBallots()
+					c.connectedEqualOrOverThreshold = true
+				}
+
 			} else {
 				c.log.Debug("validator is disconnected", "validator", v.Address(), "error", err)
+				if c.countConnectedUnlocked() < c.policy.Threshold() {
+					c.connectedEqualOrOverThreshold = false
+				}
 			}
 		}
 	}
 
 	return
 }
 
+func (c *ValidatorConnectionManager) updateBallots() {
+	ballots := c.getBallots()
+	if len(ballots) == 0 {
+		c.log.Error(
+			"get ballots, but empty",
+			"len", len(ballots),
+		)
+		return
+	}
+
+	c.log.Debug(
+		"get ballots",
+		"len", len(ballots),
+		"ballot", ballots,
+	)
+
+	for _, blt := range ballots {
+		encoded, _ := blt.Serialize()
+		c.network.MessageBroker().Receive(common.NewNetworkMessage(common.BallotMessage, encoded))
+	}
+	return
+}
+
+func (c *ValidatorConnectionManager) getBallots() map[string]ballot.Ballot {
+	c.RLock()
+	defer c.RUnlock()
+
+	ballots := map[string]ballot.Ballot{}
+	for addr, connected := range c.connected {
+		var validator *node.Validator
+		if validator = c.localNode.Validator(addr); validator == nil {
+			continue
+		}
+		if validator.Address() == c.localNode.Address() {
+			continue
+		}
+		if validator.Endpoint() == nil {
+			continue
+		}
+
+		if !connected {
+			continue
+		}
+
+		func(v *node.Validator) {
+			client := c.GetConnection(v.Address())
+
+			var err error
+			var retBody []byte
+			retBody, err = client.GetBallots()
+			if err != nil {
+				c.log.Error(
+					"failed to get ballots",
+					"error", err,
+					"validator", v.Address(),
+					"response", string(retBody),
+				)
+			}
+			if len(retBody) == 0 {
+				return
+			}
+			sc := bufio.NewScanner(strings.NewReader(string(retBody)))
+			for sc.Scan() {
+				var itemType api.NodeItemDataType
+				var b interface{}
+				itemType, b, err = api.UnmarshalNodeItemResponse(sc.Bytes())
+				if err != nil {
+					c.log.Error(
+						"failed to unmarshal getting response body",
+						"error", err,
+						"validator", v.Address(),
+						"itemType", itemType,
+						"response", string(retBody),
+						"result", b,
+					)
+				}
+
+				if itemType != api.NodeItemBallot {
+					continue
+				}
+
+				blt, ok := b.(ballot.Ballot)
+				if !ok {
+					continue
+				}
+
+				ballots[blt.GetHash()] = blt
```

Comment:
> We can filter the `Ballot`, which is already finished in current node.

@spikeekips I don't understand exactly what you mean.
This ballots `map` is for filtering same ballots because these are from all validators.

Reply:
@Charleslee522 I mean, the recieved  ballot can be for old block or invalid, I know this kind of filtering can be done by the next checker.

Reply:
@spikeekips Yes, exactly. Do you still think we should check ballot validity in this point?
I figure it's enough to check it in checker.

Reply:
@Charleslee522 Not at this time, but instead of entire validation for a ballot, checking the block height and hash is valid will be enough I think.

Reply:
@spikeekips I think that kind of work can make the `sebak` efficient, but this filtering is consensus's role.
And I don't think using consensus module in network is not easy because of recursive import.

Reply:
@Charleslee522 Okay, I see :)